packages/lockers/src/integrations/yieldbasis/YbGovVoter.sol

```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.28;
+
+// Cleaned up imports: Removed YieldBasisLocker as it was not used
+import {YieldBasisProtocol, YieldBasisLocker} from "@address-book/src/YieldbasisEthereum.sol";
+import {IYbPlugin, Tally} from "src/interfaces/IYbPlugin.sol";
+import {VoterBase} from "src/VoterBase.sol";
+
+/// @notice Structure representing a single vote operation
+struct Vote {
+    uint256 _voteId;         // The ID of the proposal
+    Tally _tally;            // The voting choice (e.g., Yes, No, Abstain)
+    bool _tryEarlyExecution; // Flag to attempt execution immediately if the vote passes
+}
+
+/// @title YbGovVoter
+/// @notice A Safe Module designed to batch votes for the YieldBasis governance protocol
+/// @dev Inherits from VoterBase to handle Safe access control and permissions
+contract YbGovVoter is VoterBase {
+
+    /// @notice The immutable address of the YieldBasis Plugin contract
+    address public immutable PLUGIN = YieldBasisProtocol.PLUGIN;
+
+    /// @notice YB Locker
+    address public immutable LOCKER = YieldBasisLocker.LOCKER;
+
+    /// @notice Constructor to initialize the voter contract
+    /// @param _gateway The address of the governance gateway or Safe module controller
+    constructor(address _gateway) VoterBase(_gateway) {}
+   
+    /// @notice Executes a batch of votes on the YieldBasis Plugin
+    /// @dev This function is atomic: if one vote fails, the entire transaction reverts.
+    ///      Protected by 'hasProposalsOrAllPermission' from VoterBase.
+    /// @param _votes An array of Vote structs containing the vote parameters
+    function votes(Vote[] calldata _votes) external hasProposalsOrAllPermission {
+        uint256 length = _votes.length;
+
+        // Iterate through the votes array
+        for (uint256 i; i < length;) {
```

Comment:
This is not important, especially considering the current market conditions, but in fact reading directly from the calldata layout is cheaper than pushing the value into the stack before manipulating it. We are talking about saving something like ~12 gas here. This comment is nothing more than a good to know thing.

```suggestion
        // Iterate through the votes array
        for (uint256 i; i < _votes.length;) {
```